lib/bulk_processor/back_end/active_job.rb

```diff
@@ -1,20 +1,25 @@
+require 'active_job'
+
+require_relative 'active_job/process_csv_job'
+require_relative 'active_job/split_csv_job'
+
 class BulkProcessor
   module BackEnd
     # Execute jobs via ActiveJob, e.g. Resque
     class ActiveJob
       def initialize(processor_class:, payload:, key:)
-        @processor_class = processor_class
-        @payload = payload
+        @processor_class = processor_class.name
+        @payload = PayloadSerializer.serialize(payload)
         @key = key
       end
 
       def start
-        Job::ProcessCSV.perform_later(processor_class.name, payload, key)
+        ActiveJob::ProcessCSVJob.perform_later(processor_class, payload, key)
       end
 
       def split(num_processes)
-        Job::SplitCSV.perform_later(processor_class.name, payload,
-                                    key, num_processes)
+        ActiveJob::SplitCSVJob.perform_later(processor_class, payload, key,
+                                             num_processes.to_s)
```

Comment:
Why the `to_s`? An integer is the right type here, and it seems like `SplitCSVJob.perform_later` should be responsible for making it a string if it needs to.